### PR TITLE
ci: Re-add configs removed in cmake migration

### PR DESCRIPTION
**Pull request by maflcko**
In commit 9730288a0cd3f33021ef00fb2d95e5216d10ab61 many configs were removed from the CI without explanation.

Fix it by adding them back.

Can be reviewed by looking at:

* the parity table https://gist.github.com/hebasto/2ef97d3a726bfce08ded9df07f7dab5e
* the installed packages
* the CI logs from before the cmake migration and the CI logs of this pull request

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -72,7 +72,7 @@ jobs:
         run: |
           # Run tests on commits after the last merge commit and before the PR head commit
           # Use clang++, because it is a bit faster and uses less memory than g++
-          git rebase --exec "echo Running test-one-commit on \$( git log -1 ) && CC=clang CXX=clang++ cmake -B build -DBUILD_GUI=ON -DBUILD_BENCH=ON -DBUILD_FUZZ_BINARY=ON -DWITH_BDB=ON -DWITH_NATPMP=ON -DWITH_MINIUPNPC=ON -DWITH_USDT=ON && cmake --build build -j $(nproc) && ctest --test-dir build -j $(nproc) && ./build/test/functional/test_runner.py -j $(( $(nproc) * 2 ))" ${{ env.TEST_BASE }}
+          git rebase --exec "echo Running test-one-commit on \$( git log -1 ) && CC=clang CXX=clang++ cmake -B build -DWITH_ZMQ=ON -DBUILD_GUI=ON -DBUILD_BENCH=ON -DBUILD_FUZZ_BINARY=ON -DWITH_BDB=ON -DWITH_NATPMP=ON -DWITH_MINIUPNPC=ON -DWITH_USDT=ON && cmake --build build -j $(nproc) && ctest --test-dir build -j $(nproc) && ./build/test/functional/test_runner.py -j $(( $(nproc) * 2 ))" ${{ env.TEST_BASE }}
```
Comment (fanquake) [inline .github/workflows/ci.yml:75]: Should this job also be installing `libzmq3-dev` ?

**Comment by fanquake** [comment]
> many configs were removed from the CI without explanation.

Nice catch. It's not clear why a bunch of CI configs would have been removed..

**Comment by hebasto** [comment]
> > many configs were removed from the CI without explanation.
> 
> Nice catch. It's not clear why a bunch of CI configs would have been removed..

Sorry for overlooking them.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (fanquake) [reply .github/workflows/ci.yml:75]: Sure, it's likely fine, just inconsistent with the other CIs.

**Review by fanquake — APPROVED** [review]
ACK fa80d39d82013b21a70677ac79bdff8c614379b7

**Review by hebasto — COMMENTED** [review]
Post-merge ACK fa80d39d82013b21a70677ac79bdff8c614379b7.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (fanquake) [reply .github/workflows/ci.yml:75]: Looks like this doesn't work, given: https://github.com/bitcoin/bitcoin/actions/runs/10612371177/job/29413871228?pr=30749 ?

Comment (fanquake) [reply .github/workflows/ci.yml:75]: No worries, I've pushed a commit into #30749.

**Pull request merged**